scripts/stubsabot.py

```diff
@@ -172,11 +172,12 @@ async def release_contains_py_typed(release_to_download: PypiReleaseDownload, *,
 
 async def find_first_release_with_py_typed(pypi_info: PypiInfo, *, session: aiohttp.ClientSession) -> PypiReleaseDownload | None:
     release_iter = pypi_info.releases_in_descending_order()
+    latest_release = next(release_iter)
     # If the latest release is not py.typed, assume none are.
-    if not (await release_contains_py_typed(release := next(release_iter), session=session)):
+    if not (await release_contains_py_typed(latest_release, session=session)):
         return None
 
-    first_release_with_py_typed: PypiReleaseDownload | None = None
+    first_release_with_py_typed = latest_release
```

Comment:
if I understand correctly, we should never return pre-releases, even if the latest pre-release version happens to be the only release containing a `py.typed`.

We could filter away pre-releases very early:

```python3
release_iter = (
    release for release in pypi_info.releases_in_descending_order()
    if not release.version.is_prerelease
)
```

Reply:
> if I understand correctly, we should never return pre-releases, even if the latest pre-release version happens to be the only release containing a `py.typed`.

Yes, that's correct. Good catch.

Reply:
Oh I see where I went wrong. Sorry about that.
And nice to see even further improvements.